.travis.yml

```diff
@@ -0,0 +1,16 @@
+# vim ft=yaml
+language: python
+sudo: false
+cache:
+  directories:
+    - $HOME/.cache/pip
+
+python:
+  - 3.6
+
+before_install:
+  - python -m pip install --upgrade pip
+  - pip install "flake8<3.0" flake8-putty
```

Comment:
Might be worth adding a `requirements-dev.txt` file, specifying development requirements and installing here from that file, using `pip install -r requirements-dev.txt`?

Reply:
With the transition to [PEP508](https://www.python.org/dev/peps/pep-0508/) and others influencing the `setup.cfg` & setuptools mode of installation it is possible to minimize different distribution configurations - this covers developers and contributors if they install with ``pip install -e .[all]``.

For CI you are sometimes interested in avoiding the installation of the package (via pip, setup.py or wheels). We can try this and if it becomes burdensome to handle environments for Circle and/or Travis, then include a `.maintenance/requirements-dev.txt` file. WDYT?